javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/TypeTransformerImpl.java

```diff
@@ -22,7 +23,9 @@ final class TypeTransformerImpl implements TypeTransformer {
     this.agentBuilder = agentBuilder;
     adviceMapping =
         Advice.withCustomMapping()
-            .with(new Advice.AssignReturned.Factory().withSuppressed(Throwable.class));
+            .with(
+                new ForceDynamicallyTypedAssignReturnedFactory(
+                    new Advice.AssignReturned.Factory().withSuppressed(Throwable.class)));
```

Comment:
[for reviewer] this allows to automatically apply dynamic typing when advices are inlined (thanks @JonasKunz for finding it).

Reply:
Actually I removed this not too long ago figuring that in inline advice we won't usually need it. Adding it back is fine, I guess I was just naively hoping that we won't have too many cases where we need to resort to Object[]. Sorry about that.